src/VecSim/algorithms/brute_force/brute_force.h

```diff
@@ -35,49 +34,21 @@ class BruteForceIndex : public VecSimIndex {
     virtual VecSimBatchIterator *newBatchIterator(const void *queryBlob,
                                                   VecSimQueryParams *queryParams) override;
     bool preferAdHocSearch(size_t subsetSize, size_t k, bool initial_check) override;
-    inline labelType getVectorLabel(idType id) const {
-        return idToLabelMapping.at(id);
-    } // throws out_of_range
-    inline labelType getVectorId(labelType label) const {
-        return labelToIdLookup.at(label);
-    } // throws out_of_range
+    virtual inline labelType getVectorId(labelType label) const = 0;
+    inline labelType getVectorLabel(idType id) const { return idToLabelMapping.at(id); }
 
     inline vecsim_stl::vector<VectorBlock *> getVectorBlocks() const { return vectorBlocks; }
-    inline Spaces::dist_func_ptr_ty<float> distFunc() const { return dist_func; }
-    inline void setLastSearchMode(VecSearchMode mode) override { this->last_mode = mode; }
     virtual ~BruteForceIndex();
 
-private:
+protected:
     void updateVector(idType id, const void *vector_data);
```

Comment:
maybe this should be an API function? for the case that in multi index you are given a label that is already exist but we dont want to add another vector, but actually update?

Reply:
There is no use for this functionality at the moment. We might add this API in the future as an optimization.